### PR TITLE
Fix coqdoc section titles

### PR DESCRIPTION
**Pull request by jdchristensen**
I realized that I was using things like `(** * text here *)` incorrectly,  causing headings to be much too long in the generated html documentation, so I fixed my mistakes.  While I was at it, I found a few other places to fix.

Is there a way to preview the coqdoc html documentation before merging?

### CODE REVIEW

**Comment by Alizter** [comment]
@jdchristensen no preview unfortunately, however the built artefacts can be downloaded when the job is finished.

**Comment by jdchristensen** [comment]
@Alizter Can I build the docs locally (easily)?

**Comment by Alizter** [comment]
@jdchristensen Yes, if you want to build with Dune you can do `dune build @doc`. It will probably complain about `odoc` but just ignore that. You will find the html files in `_build/default/theories/HoTT.html/`.

**Comment by jdchristensen** [comment]
I added instructions on comments to STYLE.md.  Despite the instructions in that file, I manually updated the TOC at the top, since I don't have doctoc installed.  Hope I did it right...

**Comment by jdchristensen** [comment]
I silenced all of the warnings coqdoc was generating, which were in fact valid warnings.  Many of the pages look better now.

The only other issue I noticed is that some Category library comments use unicode, and the charset in the html files (both the toc and the individual files) isn't set to utf, so the unicode symbols appear garbled.  Is it easy to change the generated html to use unicode?

**Comment by Alizter** [comment]
@jdchristensen Don't worry about the TOC too much, we can fix it when we come across it.

**Comment by Alizter** [comment]
Regarding unicode, I think it might be possible but I'll have to look into it later. Please open an issue about it.

File: STYLE.md [hunk 1]
```diff
@@ -906,14 +906,32 @@ excessive churn in the diff.  If you wish, you can submit a separate
...
+
+```coq
+(** * This is a top-level section *)
+(** ** This is a subsection *)
+(** *** This is a sub-subsection *)
+```
+
+Section titles should be less than 80 characters, on one line, and not
+end in a period.  Other comments are generally written using the style
+`(** This is a comment. *)` or `(* This is a comment. *)`, with the
```
Comment (Alizter) [inline STYLE.md:931]: IINM the former is included in the doc but the latter is not.

Comment (jdchristensen) [reply STYLE.md:931]: Oh, you are right.  That's controlled by `--parse-comments`.  I would prefer it if all comments were shown, as the inline comments are often quite useful.

**Comment by jdchristensen** [comment]
> Regarding unicode, I think it might be possible but I'll have to look into it later. Please open an issue about it.

The docs say that you need to pass one of `-utf8` or `--utf8`.  In Makefile.coq.local-early, I see that `-utf8` is specified.  I can't see where dune is calling `coqdoc`, though.

**Comment by jdchristensen** [comment]
> @jdchristensen Don't worry about the TOC too much, we can fix it when we come across it.

I tried viewing this PR's STYLE.md on github, and the TOC link worked fine, so all is good.

**Comment by Alizter** [comment]
@jdchristensen To pass flags to Dune (>=3.8) you need to use the `coqdoc_flags` field of the `coq.theory` stanza.

**Comment by Alizter** [comment]
FTR, there is this extension on VSCode that I was using the get the TOC. I checked myself and your version is fine.

File: STYLE.md (hunk 1 above)
Comment (Alizter) [reply STYLE.md:931]: On the other hand, sometimes its useful to have comments that don't show up in the doc. For example `TODO` etc. If we have documentation then we ought to be using `(** *)`.

**Comment by jdchristensen** [comment]
> @jdchristensen To pass flags to Dune (>=3.8) you need to use the `coqdoc_flags` field of the `coq.theory` stanza.

I have verified that `make` correctly passes `-utf8`, so the html docs are good as produced by the CI.  I don't understand your instructions above.  Do you want to explain more or just push a separate fix of this yourself?

**Comment by Alizter** [comment]
@jdchristensen something like this:
```
--- a/theories/dune
+++ b/theories/dune
@@ -8,7 +8,8 @@
  (name HoTT)
  (package coq-hott)
  (modules :standard)
- (flags -noinit -indices-matter -color on))
+ (flags -noinit -indices-matter -color on)
+ (coqdoc_flags :standard --utf8))
```
You'll have to do the same for contrib and tests if you want to pass the same args (prob not).

**Comment by jdchristensen** [comment]
Currently `make` passes `-interpolate -utf8 --no-externals $(COQDOCEXTRAFLAGS)` (so it allows extra flags from the environment).

**Comment by Alizter** [comment]
@jdchristensen AFAICT we don't use that anywhere in the CI or anywhere else.

**Comment by Alizter** [comment]
Actually here is the same thing in Dune:
```diff
diff --git a/theories/dune b/theories/dune
index e033640d..b249275d 100644
--- a/theories/dune
+++ b/theories/dune
@@ -8,7 +8,8 @@
  (name HoTT)
  (package coq-hott)
  (modules :standard)
- (flags -noinit -indices-matter -color on))
+ (flags -noinit -indices-matter -color on)
+ (coqdoc_flags -interpolate -utf8 --no-externals %{env:COQDOCEXTRAFLAGS=}))

 ; TODO: Tests
```

**Comment by Alizter** [comment]
@jdchristensen Sorry if I was not clear earlier. I think you can remove `:standard` and just do the change I've mentioned above this comment.

**Comment by jdchristensen** [comment]
> %{env:COQDOCEXTRAFLAGS=}

This causes: `coqdoc: : no such file`.  With the change I just pushed, utf8 works properly.  For some reason, the produced files seem quite different from what make produces, e.g. the order of things in the toc is different, and lots more.  But I don't really feel like investigating further...

**Comment by Alizter** [comment]
@jdchristensen That's completely fine. We don't use the coqdoc feature of Dune. I opened an issue about it #1733 but didn't do anything further. You can think of it as another way to check doc locally. But the CI is the main builder of doc.

File: STYLE.md (hunk 1 above)
Comment (jdchristensen) [reply STYLE.md:931]: I searched, and we have lots of important comments written with `(* comment *)`, so I added `--parse-comments`.

**Comment by jdchristensen** [comment]
I think I'm done now!

**Review by Alizter — APPROVED** [review]

**Pull request merged**